Source/NSPathCell.m

```diff
@@ -0,0 +1,359 @@
+}
+
+- (void) setPathStyle: (NSPathStyle)pathStyle
+{
+  _pathStyle = pathStyle;
+}
+
+- (NSAttributedString *) placeholderAttributedString
+{
+  return _placeholderAttributedString; 
+}
+
+- (void) setPlaceholderAttributedString: (NSAttributedString *)string
+{
+  ASSIGNCOPY(_placeholderAttributedString, string);
+}
+
+- (NSString *) placeholderString
+{
+  return [_placeholderAttributedString string];
+}
+
+- (void) setPlaceholderString: (NSString *)string
+{
+  NSAttributedString *as = [[NSAttributedString alloc] initWithString: string];
+  [self setPlaceholderAttributedString: as];
+  RELEASE(as);
+}
+
+- (NSColor *) backgroundColor
+{
+  return _backgroundColor;
+}
+
+- (void) setBackgroundColor: (NSColor *)color
+{
+  ASSIGNCOPY(_backgroundColor, color);
+}
+
++ (Class) pathComponentCellClass
+{
+  return pathComponentCellClass;
+}
+
++ (void) setPathComponentCellClass: (Class)clz
+{
+  pathComponentCellClass = clz;
+}
+
+- (NSRect)rectOfPathComponentCell:(NSPathComponentCell *)cell 
+                        withFrame:(NSRect)frame 
+                           inView:(NSView *)view
+{
+  NSUInteger index = [_pathComponentCells indexOfObject: cell];
+  CGFloat cellWidth = (frame.size.width / (CGFloat)[_pathComponentCells count]);
+  return NSMakeRect(frame.origin.x + (cellWidth * (CGFloat)index),
+                    frame.origin.y,
+                    cellWidth,
+                    frame.size.height);
+}
+
+- (NSPathComponentCell *)pathComponentCellAtPoint:(NSPoint)point 
+                                        withFrame:(NSRect)frame 
+                                           inView:(NSView *)view
+{
+  NSUInteger c = [_pathComponentCells count];
+  NSUInteger woc = frame.size.width / c;
+  NSUInteger item = (NSUInteger)point.x / woc;
+
+  return [_pathComponentCells objectAtIndex: item];
+}
+
+- (NSPathComponentCell *) clickedPathComponentCell
+{
+  return _clickedPathComponentCell;
+}
+
+- (NSArray *) pathComponentCells
+{
+  return _pathComponentCells;
+}
+
+- (void) setPathComponentCells: (NSArray *)cells
+{
+  ASSIGNCOPY(_pathComponentCells, cells);
+}
+
+- (SEL) doubleAction
+{
+  return _doubleAction;
+}
+
+- (void) setDoubleAction: (SEL)action
+{
+  _doubleAction = action;
+}
+
+- (id) objectValue
+{
+  return _objectValue;
+}
+
+- (void) setObjectValue: (id)obj
+{
+  ASSIGN(_objectValue, obj);
+  [self setPathComponentCells:
+          [NSPathCell _generateCellsForURL: (NSURL *)_objectValue]];
+}
+
+- (NSURL *) URL
+{
+  return [self objectValue];
+}
+
+- (void) setURL: (NSURL *)url
+{
+  [self setObjectValue: url];
+}
+
+- (id<NSPathCellDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSPathCellDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+- (void) drawInteriorWithFrame: (NSRect)frame inView: (NSView *)controlView
+{
+  NSUInteger count = [_pathComponentCells count];
+
+  [super drawInteriorWithFrame: frame
+                        inView: controlView];
+  if (count > 0)
```

Comment:
We do not need to special case 0, in that case the enumerator would just return no elements.